Bricks/net/tcp/impl/posix.h

```diff
@@ -93,11 +93,51 @@ class SocketHandle : private SocketSystemInitializer {
     FromHandle(SOCKET handle) : handle(handle) {}
   };
 
-  inline SocketHandle(NewHandle) : socket_(::socket(AF_INET, SOCK_STREAM, IPPROTO_TCP)) {
+  inline SocketHandle(NewHandle, const bool disable_nagle_algorithm = kDisableNagleAlgorithmByDefault)
+      : socket_(::socket(AF_INET, SOCK_STREAM, IPPROTO_TCP)) {
     if (socket_ < 0) {
       CURRENT_THROW(SocketCreateException());  // LCOV_EXCL_LINE -- Not covered by unit tests.
     }
     BRICKS_NET_LOG("S%05d socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);\n", socket_);
+
+#ifndef CURRENT_WINDOWS
+    int just_one = 1;
+#else
+    u_long just_one = 1;
+#endif
+
+    // LCOV_EXCL_START
+    if (disable_nagle_algorithm) {
+#ifndef CURRENT_WINDOWS
+      if (::setsockopt(socket_, IPPROTO_TCP, TCP_NODELAY, &just_one, sizeof(just_one)))
+#else
+      if (::setsockopt(
+              socket, IPPROTO_TCP, TCP_NODELAY, reinterpret_cast<const char*>(&just_one), sizeof(just_one)))
+#endif
+      {
+        CURRENT_THROW(SocketCreateException());
+      }
+    }
+
```

Comment:
Swap where two lines? `// LCOV_EXCL_STOP` belongs to the section above, not below.